### PR TITLE
Adding accessor to DogSlot to Dog.cs

### PR DESCRIPTION
**Pull request by imann24**

### CODE REVIEW

File: Assets/Scripts/UI/Pickup Pup/DogSlot/DogSlot.cs [hunk 1]
```diff
@@ -24,16 +24,39 @@ protected bool hasDog
...
+			{
+				value.AssignSlot(this);
+			}
+			_dog = value;
+		}
+	}
+
 	bool hasButton {
 		get {
 			return button != null;
```
Comment (gracebarsny) [inline Assets/Scripts/UI/Pickup Pup/DogSlot/DogSlot.cs]: Both braces need to have their own line.

File: Assets/Scripts/UI/Pickup Pup/DogSlot/DogSlot.cs [hunk 2]
```diff
@@ -24,16 +24,39 @@ protected bool hasDog
...
 		}
 	}
 		
     protected DogDescriptor dogInfo;
-	protected Dog dog;
+
     protected Image[] images;
 
+	Dog _dog;
```
Comment (gracebarsny) [inline Assets/Scripts/UI/Pickup Pup/DogSlot/DogSlot.cs]: Do the child classes really not need this?

Comment (imann24) [inline Assets/Scripts/UI/Pickup Pup/DogSlot/DogSlot.cs]: They should always be using the mutator (ensures that the Dog ref is up to date)

Comment (gracebarsny) [inline Assets/Scripts/UI/Pickup Pup/DogSlot/DogSlot.cs]: Sounds good.

**Pull request merged**